### DISCUSSION TITLE
supabaseClient.auth.signIn doesn't return user session data

### DISCUSSION THREAD

**Discussion by ReactGuru007** [supabase/supabase#3769] [Questions]
hi there,

I tried the following signIn API, however it doesn't return user session.

```tsx
supabaseClient.auth
  .signIn({
    provider,
  })
  .then((data) => {
    console.log("data", data)
  })
```

in fact, I could only retrieve user session with the following code at useEffect

```tsx
React.useEffect(() => {
  const { data: listener } = supabaseClient.auth.onAuthStateChange(
    async (event, session) => {
      console.log(session)
      // setUser(session?.user ?? null)
      // setLoading(false)
    }
  )

  return () => {
    listener?.unsubscribe()
  }
}, [])
```

So there is no way to retrieve user info right after the sign-in? the reason I want to ask is that I would like to select my own `users` table to see if the user is a valid customer in the database. 

Does that mean you can only start pulling user-session information at the `onAuthStateChange`?

many thanks

**Comment by rasreee** [comment]
you can fetch the authenticated user using what's returned by `signIn` like so:

![code](https://user-images.githubusercontent.com/21317871/143318232-d6306fe3-af74-4f44-95b8-f12dd22b3631.png)

you just pass in `session.access_token` to the Auth API's `getUser` method!

^ i know it's technically still more than one step to get the auth user, but that's what i've been doing

**Reply by ghost** [reply]
This does not work. Because we're signing in using a provider, in that case Supabase doesn't return neither a user nor a session.

**Comment by silentworks** [comment]
This is because the user information isn't available until the third party provider has authenticated your user which requires a redirect to their server first for the handshake and then back to the supabase server before redirecting to your website. Hence, the session contains the user information but not the `.auth.signIn` method. This is only true for the magic link and third-party authentication.

We should probably add a note in our docs outlining this so users don't get confused. https://supabase.com/docs/reference/javascript/auth-signin#sign-in-using-third-party-providers

**Reply by ghost** [reply]
> This is because the user information isn't available until the third party provider has authenticated your user which requires a redirect to their server first for the handshake and then back to the supabase server before redirecting to your website. Hence, the session contains the user information but not the `.auth.signIn` method. This is only true for the magic link and third-party authentication.
> 
> We should probably add a note in our docs outlining this so users don't get confused. https://supabase.com/docs/reference/javascript/auth-signin#sign-in-using-third-party-providers

Yeah, I used the listener that is provided.
